pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -102,6 +102,51 @@ let
     else
       false;
 
+  isGccTuneSupported = tune:
+    # for x86 -mtune= takes the same values as -march, plus two more:
+    if targetPlatform.isx86 then
+      {
+        generic = true;
+        intel = true;
+      }.${tune} or (isGccArchSupported tune)
+    # on arm64, the -mtune= values are specific processors
+    else if targetPlatform.isAarch64 then
+      (if isGNU then
+        {
+          cortex-a53              = versionAtLeast ccVersion "4.8";  # gcc 8c075f
+          cortex-a72              = versionAtLeast ccVersion "5.1";  # gcc d8f70d
+          "cortex-a72.cortex-a53" = versionAtLeast ccVersion "5.1";  # gcc d8f70d
+        }.${tune} or false
+       else if isClang then
+         {
+           cortex-a53             = versionAtLeast ccVersion "3.9"; # llvm dfc5d1
+         }.${tune} or false
+       else false)
+    else if targetPlatform.isPower then
+      # powerpc does not support -march
+      true
```

Comment:
Does that mean PPC will accept all mtune flags? That doesn't seem right to me.

Reply:
> Does that mean PPC will accept all mtune flags?

Not quite; it means that nixpkgs *won't filter out* any `gcc.tune` values, and will pass them along as `-mtune=` flags.  If they are bogus, the build will fail.

The flag-filtering functions deal with two situations:

- A flag is accepted by newer versions of gcc but rejected by older versions that are still present in the bootstrap-files.
- A flag is accepted by gcc but rejected by clang

This also lets us avoid having to hand-maintain a table with every possible tuning flag.  We only track the ones that are accepted by one compiler but rejected by a different one, in situations where we care about both compilers.

This isn't really clear from the code, either from my PR or the preexisting `isGccArchSupported`, so I'll push an additional commit that adds the reasoning above as a comment.

Reply:
Please see 252a8a5bf801 -- is that more clear?

Reply:
Yes indeed. We might actually want to use this option for aarch64-darwin because clang-11 doesn't know the apple-m1 but newer versions do.